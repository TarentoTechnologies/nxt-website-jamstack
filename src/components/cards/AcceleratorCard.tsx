import { useState } from "react";

import { lightNeutralGray } from "../../styles/style-guide/ColorGuide.module.css";
import { cardHeader } from "../../styles/style-guide/Typography.module.css";
import {
  cardHover,
  cardLogo,
  cardOneImg,
  cardSize,
} from "./AcceleratorCard.module.css";

interface AcceleratorCardProps {
  title: string;
  description: any;
  logoLightSrc: string;
  logoDarkSrc: string;
  bgImg: string;
}

export const AcceleratorCard = ({
  title,
  description,
  logoDarkSrc,
  logoLightSrc,
  bgImg,
}: AcceleratorCardProps) => {
  const [hovered, setHovered] = useState(false);

  const handleMouseEnter = () => {
    setHovered(true);
  };
  const handleMouseLeave = () => {
    setHovered(false);
  };

  return (
    <div
      className="col-xs-12 col-sm-4 col-md-4 col-lg-4 mb-4"
      onMouseEnter={handleMouseEnter}
      onMouseLeave={handleMouseLeave}
    >
      {hovered ? (
        <div className={`${cardSize} ${cardHover}`}>
          <div>
            <p className={cardHeader}>{title}</p>
            <p className={lightNeutralGray}>{description}</p>
            <img
              className={cardLogo}
              src={logoLightSrc}
              alt={title}
              title={title}
            />
          </div>
        </div>
      ) : (
        <div
          className={`${cardSize} ${cardOneImg}`}
          style={{ backgroundImage: `url(${bgImg})` }}
        >
          <p className={cardHeader}>{title}</p>
          <img
            className={cardLogo}
            src={logoDarkSrc}
            alt={title}
            title={title}
          />
        </div>
      )}
    </div>
  );
};
