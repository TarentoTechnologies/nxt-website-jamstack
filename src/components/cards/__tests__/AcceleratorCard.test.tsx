import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import React from "react";

import { AcceleratorCard } from "../AcceleratorCard";

describe("Test if the AcceleratorCard component is working as expected", () => {
  it("<AcceleratorCard /> matches snapshot", () => {
    const component = render(
      <AcceleratorCard
        data-testid="accelerator-card"
        title={"BOLT - Data Platform"}
        description={
          "Data engineering at scale. Validate, transform and enrich your data for high quality data usecases."
        }
        logoLightSrc={
          "https://strapi-stage-nxt.tarento.com/uploads/bolt_light_f208259cd2.svg"
        }
        logoDarkSrc={
          "https://strapi-stage-nxt.tarento.com/uploads/bolt_dark_c35a2582e8.svg"
        }
        bgImg={
          "https://strapi-stage-nxt.tarento.com/uploads/data_platform_593cf8006d.svg"
        }
      />
    );
    expect(component.container).toMatchSnapshot();
  });

  it("<AcceleratorCard /> renders without exploding", () => {
    const component = render(
      <AcceleratorCard
        data-testid="accelerator-card"
        title={"BOLT - Data Platform"}
        description={
          "Data engineering at scale. Validate, transform and enrich your data for high quality data usecases."
        }
        logoLightSrc={
          "https://strapi-stage-nxt.tarento.com/uploads/bolt_light_f208259cd2.svg"
        }
        logoDarkSrc={
          "https://strapi-stage-nxt.tarento.com/uploads/bolt_dark_c35a2582e8.svg"
        }
        bgImg={
          "https://strapi-stage-nxt.tarento.com/uploads/data_platform_593cf8006d.svg"
        }
      />
    );
    expect(component.getByTestId("accelerator-card")).toBeInTheDocument();
  });
});
