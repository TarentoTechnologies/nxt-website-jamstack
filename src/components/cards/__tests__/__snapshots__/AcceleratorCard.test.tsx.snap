// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if the AcceleratorCard component is working as expected <AcceleratorCard /> matches snapshot 1`] = `
<div>
  <div
    class="col-xs-12 col-sm-12 col-md-4 col-lg-4 mb-4"
    data-testid="accelerator-card"
  >
    <div
      class="cardSize cardOneImg"
      data-aos="fade-up"
      style="background-image: url(https://strapi-stage-nxt.tarento.com/uploads/data_platform_593cf8006d.svg);"
    >
      <p
        class="cardHeader"
        data-aos="fade-up"
      >
        BOLT - Data Platform
      </p>
      <img
        alt="BOLT - Data Platform"
        class="cardLogo"
        data-aos="fade-up"
        src="https://strapi-stage-nxt.tarento.com/uploads/bolt_dark_c35a2582e8.svg"
        title="BOLT - Data Platform"
      />
    </div>
  </div>
</div>
`;
